cal/cal/templates/home_logged_in.html

```diff
@@ -72,15 +72,54 @@
 
   <!-- Lists and other info on the side -->
   <div class="col-md-4">
-    <div class="cal-category-list" ng-controller="CategoriesCtrl">
+    <div ng-controller="CategoriesCtrl as $ctrl">
       <h4 class="well">Categories - Last Week</h4>
-      <ul class="list-group">
-        <li class="list-group-item" ng-repeat="category in categories">
-          {% verbatim %} {{ category.label }}: {{ category.hours }} hours {% endverbatim %}
-        </li>
-      </ul>
-      <nvd3 options="options" data="categories"></nvd3>
+        <table class="table table-hover">
+          <thead>
+            <th>Label</th>
+            <th>Hours</th>
+          </thead>
+          <tbody>
+            <tr ng-repeat="category in categories">
+              {% verbatim %}
+              <td ng-hide="category.editing">
+                <input type="checkbox" ng-model="category.include" ng-click="$ctrl.chartToggle(category.id)"> {{ category.label }}
+              </td>
+              <td ng-hide="category.editing">
+                {{ category.hours }}
+              </td>
+              {% endverbatim %}
+              <td ng-show="category.editing" class="textbox">
+                <input type="text" ng-model="category.newLabel">
+              </td>
+              <td ng-show="category.editing"></td> <!-- Hacky way of making it even -->
+              <td class="text-right text-nowrap" ng-show="category.editing">
+                <button class="btn btn-success">
+                  <span class="glyphicon glyphicon-ok" ng-click="$ctrl.submit(category.id)"></span>
+                </button>
+                <button class="btn btn-warning">
+                  <span class="glyphicon glyphicon-remove" ng-click="$ctrl.cancelEdit(category.id)"></span>
+                </button>
+              </td>
+              <td class="text-right text-nowrap" ng-hide="category.editing">
+                <button class="btn btn-info">
+                  <span class="glyphicon glyphicon-edit" ng-click="$ctrl.startEdit(category.id)"></span>
+                </button>
+                <button class="btn btn-danger">
+                  <span class="glyphicon glyphicon-trash" ng-click="$ctrl.remove(category.id)"></span>
+                </button>
+              </td>
+            </tr>
+            <tr>
+              <td class="text-nowrap">
+                <button class="btn btn-success">Generate Categories</button>
+              </td>
+            </tr>
+          </tbody>
+        </table>
+      <nvd3 options="options" data="categoriesForChart"></nvd3>
```

Comment:
Instead of maintaining a whole new array of categories in memory just for the chart, you can use a `filter` angular filter by `include` here. [Link to docs](https://docs.angularjs.org/api/ng/filter/filter)